lib/mongo_client.js

```diff
@@ -848,27 +848,52 @@ var connect = function(self, url, options, callback) {
     );
   }
 
-  // Parse the string
-  var object = parse(url, options);
-  var _finalOptions = createUnifiedOptions({}, object);
-  _finalOptions = mergeOptions(_finalOptions, object, false);
-  _finalOptions = createUnifiedOptions(_finalOptions, options);
-
-  // Check if we have connection and socket timeout set
-  if (_finalOptions.socketTimeoutMS == null) _finalOptions.socketTimeoutMS = 360000;
-  if (_finalOptions.connectTimeoutMS == null) _finalOptions.connectTimeoutMS = 30000;
-
-  if (_finalOptions.db_options && _finalOptions.db_options.auth) {
-    delete _finalOptions.db_options.auth;
-  }
+  parse(url, options, function(err, object) {
+    // Do not attempt to connect if parsing error
```

Comment:
let's add one of these here, so we can make it easier to use if we potentially export this method for e.g. mongoose:

```
if (typeof options === 'function') (callback = options), (options = {});
options = options || {};
```

Reply:
oops wrong place for this, the comment should go on url_parser.js:11, sorry.